### ISSUE TITLE
[bug] Plugin: can't add anything in the head area

### ISSUE THREAD

**Issue by jcodewalker** [vdm-io/Joomla-Component-Builder#482]
### Steps to reproduce the issue
Click on "Add Custom Class Header" -> Yes
Wait for the default "use Joomla\CMS\Application\CMSApplication;" 
and "use Joomla\CMS\Plugin\CMSPlugin;" statements to appear, then try to save as is.

### Expected result
The plugin saved

### Actual result
Alert message "Some values are unacceptable" and a page error message "Invalid field:  Class Head".

### System information (as much as possible)
- OS Name & Version: 
- MySql Version: 
- Apache Version: 
- PHP Version: 
- Joomla Version: 
- JCB Version: 2.10.1
- Browser: Firefox / Chrome

### Additional comments
Also tried to follow the tutorial (https://www.youtube.com/watch?v=bs6LstIg5H0&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=87&t=448s) but it doesn't allow me to save the document...

**Comment by Llewellynvdm** [comment]
Please give us some screen shots, as I just did the same with no errors so I can not currently duplicate the issue.

**Comment by Llewellynvdm** [comment]
In fact I have many plugins and all with custom headers... and all worked first time. But to make sure I did a blank install and a still I did not get the same error/issue you explained.

**Comment by jcodewalker** [comment]
This is a fresh new plugin, no methods, nothing, just click "Yes" on custom header:
![Schermata 2019-10-11 alle 20 19 21](https://user-images.githubusercontent.com/49115855/66675011-b3446700-ec64-11e9-9392-957e2f861b96.png)
![Schermata 2019-10-11 alle 20 19 37](https://user-images.githubusercontent.com/49115855/66675026-b93a4800-ec64-11e9-8170-74acb5ede19d.png)
What test do you want I do? I can't send you my computer (BTW, it is in localhost if this matter)

**Comment by jcodewalker** [comment]
So, it seems to be the same issue of CSS with codemirror, do you remember?
I'm talking about this: https://github.com/vdm-io/Joomla-Component-Builder/issues/456

If I hack the form and set the field as "text" I can add whatever I want and then, when I revert it back to editor, it works perfectly. But, if I click on radio button "No" (hiding the editor field), save it, and then click again on "Yes" it doesn't work again.

It seems like codemirror adds some strange character, not visible, that interferes. Infact, if you see the attached image of my prev post, you can see the "use" statements are not on top of the editor field. Then even if I delete that space it doesn't work. No matter what I do.

If the form is loaded with the editor already open, even if the field is empty, it works fine, even if I save it still empty. The issues come up when I want to open the editor by clicking on the radio button with codemirror active.

You fixed the CSS issue, so I hope this help you to understand the solution.

Thanks in advance.

**Comment by Llewellynvdm** [comment]
I personally have never used code-mirror, so I have never had these issues.

But okay, I have checked the field, and as you can [see here it is not set as required](https://github.com/vdm-io/Joomla-Component-Builder/blob/staging/admin/models/forms/joomla_plugin.xml#L215) we also added the JCB editor [switch to the view as you see here](https://github.com/vdm-io/Joomla-Component-Builder/blob/staging/admin/models/joomla_plugin.php#L420) so the fact that it makes the field red, and gives an error is all Joomla as this should not happen. JCB does not control the actual field behavior we just setup the xml and give it to the Joomla API, which in turn builds the field and set all the script needed.

So in conclusion this is not a JCB error, but a code-mirror issue. I know this is a pain, and if I could I would resolve it. I you have the time, take up the issue with Joomla, and the code-mirror plugin for Joomla.

If they can give us more insight it will be helpful.

Until then I will close this issue, but you are most welcome to share any feedback you may find.

**Comment by jcodewalker** [comment]
First of all is not that I fallen in love with code-mirror… what editor do you use? If there is one that works fine please tell us.

**Comment by Llewellynvdm** [comment]
No, I did not think you loyal to code-mirror... but it is nice to have something where tabs work, and little bit of code highlighting.

I don't use any... I use none :+1: so you can imagine by now I have become affluent in the task of coding with no code highlighting or tabs. It has its benefits in terms of discipline, and copy pasting of tabs becomes second nature. But it is a pain and not recommended as time will proof that tools like code-mirror are valuable assets.

I tried [Rokpad](https://rockettheme.com/joomla/extensions/rokpad) a few times, but have not found it practical with php, so since I do better without it, I just moved back to none.  But you can give it a try and let me know if it works better.

Honestly I think code-mirror is ideal for the task (much bigger community around it), but it has these  quirks issues that in my mind should net be happening.
